### ISSUE TITLE
The copy whole process indicator doesn't show changes

### ISSUE THREAD

**Issue by metim** [mucommander/mucommander#68]
While copying several files as one operation the indicator of whole completion status doesn't show changes (it stays at 0% untill the end of the process).
Thanks.

**Comment by ahadas** [comment]
Right, but what would you propose to do when at least one of the copied files is a directory? it may take time to compute its size (imagine a directory with many sub-directories and many files)

**Comment by metim** [comment]
I don’t know but many filemanagers managed to solve this (trolcommander solved it as well). Maybe create a separate thread which will calculate the size while the first one performs copying and after getting size update values. Otherwise it is difficult to copy big dirs without seeing how long will it take.

**Comment by ahadas** [comment]
The approach you propose can lead to 'go back in time' effect, that is when the progress becomes lower than it was before. And it will never be 100% correct since one can write into one of the sub-folders after you examined for its size but before this sub-folder is being copied.

I doubt other file-manager have a good solution for this that is not based on some heuristic . It is a complicated problem - do you remember Microsoft's progress bar that was presented in the installation of Windows that was going from 0 to almost 100% in just a few minutes and then getting stuck in that state for much more than you would expect? :)

May I ask why is that difficult to copy big directories without seeing how long it will take? would you still find it necessary if the next release will provide proper support for executing operations like copying in the background?

**Comment by metim** [comment]
I understand that there may be many problems with how to calculate correct time, but it is only approximation and IMHO nobody will anticipate exact results (especially after manually changing the dir while copying is in process). 

As of why I find it usefull. For example if I copy/move big dirs on my hard drive – some programs on my PC can become slower (if they depend on reading/writing data) so I want to know when I can use them again. 
Another example: My brother asks me to copy a large dir to his phone/flash drive. I say “OK, come again in 15 minutes” or even “wait it will take only 1 minute”. Otherwise I have to manually look for size of the dir, then think about the copy speed of his device (which may not be obvious due to different driver realizations) and then calculate the answer.
